src/Quartz/Simpl/PropertySettingJobFactory.cs

```diff
@@ -135,9 +135,16 @@ protected virtual IJob InstantiateJob(TriggerFiredBundle bundle, IScheduler sche
     /// <param name="data">The data to set.</param>
     public virtual void SetObjectProperties(object obj, JobDataMap data)
     {
-        foreach (string name in data.Keys)
+        if (obj is IJobWrapper jobWrapper)
```

Comment:
If we have a IJobWrapper here, we need to drill down until we get to the baseTarget and set the Properties there.

This removes the need for overriding SetObjectProperties for inheritors using the IJobWrapper interface.